konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/data/JData.java

```diff
@@ -326,17 +327,20 @@ public void putList(String key, List<?> data, ValueType vt){
 
     @Override
     public boolean hasMetaData() {
-        return metaData != null;
+        return dataMap.containsKey(Data.RESERVED_KEY_METADATA);
     }
 
     @Override
     public Data getMetaData() {
-        return metaData;
+        DataValue metaData = (DataValue) dataMap.get(Data.RESERVED_KEY_METADATA);
+        if (metaData == null)
+            return null;
+        return metaData.get();
     }
 
     @Override
     public void setMetaData(Data data) {
-        this.metaData = data;
+        dataMap.put(Data.RESERVED_KEY_METADATA, new DataValue(data));
```

Comment:
Please revert these changes
Storing in protobuf like this is totally fine, but not at Java level. Let's keep the metaData field.
To do it this way will required filtering the keys to remove the metadata key for methods like keys(), etc.

Reply:
Appears to be not so small in protobuf ser-de. Fixed.